dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -433,6 +433,11 @@ private void updateInfo(Tool originalTool, Tool newTool) {
             originalTool.setTopicSelection(newTool.getTopicSelection());
         }
 
+        // If the user chose an AI topic, it means that they have reviewed and approved it
+        if (newTool.getTopicSelection() == TopicSelection.AI) {
```

Comment:
Hmmmm, is this the general update code?
What if the user just means to update the forum url for example and did not mean to approve the current ai topic?

Reply:
Hmm good point. My idea was that if the user selected AI topic then it automatically means that they have reviewed and approved it, so it's kind of redundant to pass along `approvedAITopic`. But in your scenario, it can lead to accidental approval.

Kind of makes me want to separate out the topic change updates to its own endpoint :thinking:

Reply:
That could work too, but I don't think it is (always) redundant to pass along approvedAITopic, I think you have a few scenarios

a) User means to change something else entirely (selection stays the same as before, approved topic doesn't change) 
b) User means to approve the AI topic (selection stays, boolean changes to true)
c) User means to change to .dockstore.yml topic (selection changes, boolean is now redundant)
d) User changes back to AI topic (selection changes, boolean goes back to true and is not redundant)

In other words, the boolean is redundant if the topic is not AI generated.